### PR TITLE
adds common folders for particle tools

### PR DESCRIPTION
**Pull request by m-mcgowan**
This is a quick implementation based on the work discussed so far. 

Happy for thoughts on structure, names etc..

- currently all folders are in `Projects`. should it be split into `Projects`, `Libraries` and `Particle` so each region has it's own class? (with Particle defining the shared particle folder, used by the others.)

- are the names ok? should projectsFolder() be communityProjectsFolder(), or importedProjectsFolder() etc...

### CODE REVIEW

File: src/cmd/projects.js [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+export class Projects {
+
+	userHomeFolder() {
+		return os.homedir();
+	}
+
+	// todo - documnents folder?
+
+	particleFolder() {
+		return path.join(this.userHomeFolder(), particle);
```
Comment (monkbroc) [inline src/cmd/projects.js]: I'm going to look for an API to get the Windows "Documents" folder.

**Comment by monkbroc** [comment]
I really think we ought to use the Windows "Documents" folder as this is what people would expect. I'll look for a Node.js way to get that.

**Comment by monkbroc** [comment]
The only way I found to get the path of the Documents folder is to query the registry.

Would you be willing to pull in winreg as a dependency? It uses `reg.exe` to do registry queries so it doesn't need any compiled dependencies unlike some other modules. It's just a little big.
https://github.com/fresc81/node-winreg

Here's the winreg example modified to print the "Documents" path. I tried it on Windows 10.
```
var winreg = require('winreg');
var regKey = winreg({
      hive: winreg.HKCU,                                        // open registry hive HKEY_CURRENT_USER
      key:  '\\Software\\Microsoft\\Windows\\CurrentVersion\\Explorer\\Shell Folders' // key containing user folder paths
    });

// list "Documents" folder path
regKey.values(function (err, items) {
  if (err) {
    console.log('ERROR: '+err);
  } else {
    for (var i=0; i<items.length; i++) {
      if (items[i].name === "Personal") {
        console.log(items[i].value);
      }
    }
  }
});
```

File: src/cmd/projects.js [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+			return promise;
+		}
+	});
+}
+
+
+export class Projects {
+
+	userHomeFolder() {
+		return os.homedir();
```
Comment (m-mcgowan) [inline src/cmd/projects.js]: I was assuming Babel added the necessary polyfill along with all the other newer node functions. Does it not do that?

**Comment by m-mcgowan** [comment]
>  It's just a little big.
A little big? it's 25k source...and no runtime dependencies...I guess you're pulling my leg? ;-)  I'm happy going with that. Seems to be the expected behaviour for the platform.

**Comment by monkbroc** [comment]
Yeah, sure. I guess I was expecting a 50 line library wrapping running `reg.exe` in `child_process`, not a 1000 line beast.

**Pull request merged**